src/player.js

```diff
@@ -1837,33 +1985,47 @@ export default class Player extends FakeEventTarget {
   }
 
   /**
-   * Handles auto play.
+   * Handles auto pause.
    * @returns {void}
    * @private
    */
-  _handleAutoPlay(): void {
-    if (this._config.playback.autoplay === true) {
-      const allowMutedAutoPlay = this._config.playback.allowMutedAutoPlay;
-      Player.getCapabilities(this.engineType).then(capabilities => {
-        if (capabilities.autoplay) {
-          onAutoPlay();
-        } else {
-          if (capabilities.mutedAutoPlay) {
-            if (this.muted && !this._fallbackToMutedAutoPlay) {
-              onMutedAutoPlay();
-            } else if (allowMutedAutoPlay) {
-              onFallbackToMutedAutoPlay();
-            } else {
-              onAutoPlayFailed();
-            }
-          } else {
-            onAutoPlayFailed();
+  _initAutoPause(): void {
+    if (this._config.playback.autopause === true) {
+      this._mediaEventManager.listen(this, CustomEventType.VISIBILITY_CHANGE, (e: FakeEvent) => {
+        if (!e.payload.visible) {
+          if (!this._isFloating && !this.isInPictureInPicture() && this._playbackStart && !this.paused) {
+            this.pause();
+            this._autoPaused = true;
```

Comment:
I would reset it in playing for when the play and pause called by API
(paused on invisible => played by api => paused by api => become visible)

Reply:
Not relevant anymore